modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/service/test/SegmentsExperimentLocalServiceTest.java

```diff
@@ -674,43 +688,140 @@ public void testUpdateSegmentsExperimentNameInStatusRunning()
-			SegmentsTestUtil.addSegmentsExperience(
-				segmentsExperience1.getGroupId(),
-				segmentsExperience1.getClassNameId(),
-				segmentsExperience1.getClassPK());
+		SegmentsExperience segmentsExperience1 = _addSegmentsExperience(0);
+		SegmentsExperience segmentsExperience2 = _addSegmentsExperience(-2);
+		SegmentsExperience segmentsExperience3 = _addSegmentsExperience(-3);
 
 		SegmentsExperiment segmentsExperiment = _addSegmentsExperiment(
-			segmentsExperience1);
+			segmentsExperience2);
 
 		SegmentsExperience variantSegmentsExperience =
-			SegmentsTestUtil.addSegmentsExperience(
-				segmentsExperience1.getGroupId(),
-				segmentsExperience1.getClassNameId(),
-				segmentsExperience1.getClassPK());
+			_addVariantSegmentsExperience(
+				segmentsExperience2.getSegmentsEntryId(),
+				segmentsExperiment.getSegmentsExperimentId());
 
-		_segmentsExperienceLocalService.updateSegmentsExperienceActive(
-			variantSegmentsExperience.getSegmentsExperienceId(), false);
+		segmentsExperiment.setStatus(
+			SegmentsExperimentConstants.STATUS_FINISHED_WINNER);
 
-		_segmentsExperimentRelLocalService.addSegmentsExperimentRel(
-			segmentsExperiment.getSegmentsExperimentId(),
-			variantSegmentsExperience.getSegmentsExperienceId(),
-			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+		_segmentsExperimentLocalService.updateSegmentsExperiment(
+			segmentsExperiment);
 
 		_segmentsExperimentLocalService.updateSegmentsExperimentStatus(
 			segmentsExperiment.getSegmentsExperimentId(),
-			SegmentsExperimentConstants.STATUS_RUNNING);
+			variantSegmentsExperience.getSegmentsExperienceId(),
+			SegmentsExperimentConstants.STATUS_COMPLETED);
+
+		segmentsExperience1 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience1.getSegmentsExperienceId());
+
+		Assert.assertTrue(segmentsExperience1.isActive());
+		Assert.assertEquals(0, segmentsExperience1.getPriority());
+
+		segmentsExperience2 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience2.getSegmentsExperienceId());
+
+		Assert.assertFalse(segmentsExperience2.isActive());
+		Assert.assertEquals(1, segmentsExperience2.getPriority());
+
+		segmentsExperience3 =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				segmentsExperience3.getSegmentsExperienceId());
+
+		Assert.assertTrue(segmentsExperience3.isActive());
+		Assert.assertEquals(-3, segmentsExperience3.getPriority());
+
+		variantSegmentsExperience =
+			_segmentsExperienceLocalService.fetchSegmentsExperience(
+				variantSegmentsExperience.getSegmentsExperienceId());
+
+		Assert.assertTrue(variantSegmentsExperience.isActive());
+		Assert.assertEquals(-2, variantSegmentsExperience.getPriority());
+	}
+
+	@Test
+	public void testUpdateSegmentsExperimentStatusToCompletedWithWinnerSegmentsExperienceAndControlSegmentsExperienceWithPriorityZero()
+		throws Exception {
+
+		SegmentsExperience segmentsExperience1 = _addSegmentsExperience(1);
```

Comment:
Better if you use directly append and add, right?, this way we now what we are trying to do, as we are only calling a method, hide it from us, doesn't give a lot of value.

Reply:
I had it this way, but the code was a bit ugly

Reply:
Now we don't know what experience you are trying to create, it is usually not a good idea to hide the way we are creating the experiences, since the way we create them have different impacts in their behaviour.

Reply:
I see your point,  I'm going to change it

Reply:
applied ;)